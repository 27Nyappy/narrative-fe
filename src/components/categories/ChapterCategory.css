.ChapterCategory {
  background-color: #ccc;
  width: 50%;
  display: flex;
  flex-direction: column;
  justify-content: center;
}

.h2 {
  align-self: center;
}

.div {
  align-self: center;
  width: 85%;
  height: 40%;
  display: flex;
  justify-content: space-between;

}