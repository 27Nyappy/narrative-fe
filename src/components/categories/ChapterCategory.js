import React from 'react';
import PropTypes from 'prop-types';
import ViewAllButton from './ViewAllButton';
import styles from './ChapterCategory.css';
import AddChapterButton from './AddChapterButton';

const ChapterCategory = ({ handleViewChapters, show, hideModal, showModal }) => {
  return (
    <section className={styles.ChapterCategory}>
      <h2>Chapters</h2>

      <div className={styles.ChapterCategoryContainer}>
        <AddChapterButton show={show} hideModal={hideModal} showModal={showModal} />
        <ViewAllButton handleViewClick={handleViewChapters} />
      </div>
    </section>
  );
};

ChapterCategory.propTypes = {
  handleViewChapters: PropTypes.func.isRequired,
  show: PropTypes.bool,
  hideModal: PropTypes.func,
  showModal: PropTypes.func
};

export default ChapterCategory;