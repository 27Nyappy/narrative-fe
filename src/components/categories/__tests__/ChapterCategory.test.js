import React from 'react';
import { shallow } from 'enzyme';
import ChapterCategory from '../ChapterCategory';

describe('ChapterCategory component', () => {
  it('renders a chapter category', () => {
    const wrapper = shallow(<ChapterCategory handleAddChapter={() => true} handleViewChapters={() => true}/>);
    expect(wrapper).toMatchSnapshot();
  });
});