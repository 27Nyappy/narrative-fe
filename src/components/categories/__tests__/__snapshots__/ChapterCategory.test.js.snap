// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChapterCategory component renders a chapter category 1`] = `
<section>
  <h2>
    Chapters
  </h2>
  <div>
    <AddButton
      handleAddClick={[Function]}
    />
    <ViewAllButton
      handleViewClick={[Function]}
    />
  </div>
</section>
`;
